packages/caver-wallet/src/index.js

```diff
@@ -16,13 +16,307 @@
+
+        founded.key = keyring.copy().key
+        return founded
+    }
+
+    /**
+     * querys the keyring address, and return.
+     *
+     * @param {string} address The address of keyring to query.
+     * @return {Keyring}
+     */
+    getKeyring(address) {
+        if (!utils.isAddress(address))
+            throw new Error(`Invalid address ${address}. To get keyring from wallet, you need to pass valid address string as a parameter.`)
+
+        const founded = this._addressKeyringMap.get(address.toLowerCase())
+
+        return founded
+    }
+
+    /**
+     * addes keyring to keyringContainer.
+     *
+     * @param {Keyring} keyring A keyring instance to add to keyringContainer.
+     * @return {Keyring}
+     */
+    add(keyring) {
+        if (this._addressKeyringMap.get(keyring.address.toLowerCase()) !== undefined)
+            throw new Error(`Duplicate Account ${keyring.address}`)
+
+        const keyringToAdd = keyring.copy()
+
+        this._length++
+        this._addressKeyringMap.set(keyringToAdd.address.toLowerCase(), keyringToAdd)
+
+        return keyringToAdd
+    }
+
+    /**
+     * deletes keyring from keyringContainer.
+     *
+     * @param {string} address An address of keyring in keyringContainer.
+     * @return {boolean}
+     */
+    remove(address) {
+        let keyringToRemove
+        if (utils.isAddress(address)) {
+            keyringToRemove = this.getKeyring(address)
+        } else {
+            throw new Error(`To remove keyring, parameter should be address string.`)
+        }
+
+        if (keyringToRemove === undefined) return false
+
+        keyringToRemove.key = null
+        this._addressKeyringMap.delete(keyringToRemove.address.toLowerCase())
+
+        this._length--
+
+        return true
+    }
+
+    /**
+     * signs with data and returns result object that includes `signsture`, `message` and `messageHash`
+     *
+     * @param {string} address An address of keyring in keyringContainer.
+     * @param {string} data The data string to sign.
+     * @param {number} [role] A number indicating the role of the key. You can use `caver.wallet.keyring.role`.
+     * @param {number} [index] An index of key to use for signing.
+     * @return {object}
+     */
+    signMessage(address, data, role, index) {
+        const keyring = this.getKeyring(address)
+        if (keyring === undefined) throw new Error(`Failed to find keyring from wallet with ${address}`)
+        return keyring.signMessage(data, role, index)
+    }
+
+    /**
+     * signs the transaction using one key and return the transactionHash
+     *
+     * @param {string} address An address of keyring in keyringContainer.
+     * @param {Transaction} transaction A transaction object.
+     * @param {number} [index] An index of key to use for signing.
+     * @param {function} [hasher] A function to return hash of transaction.
+     * @return {string}
+     */
+    async signWithKey(address, transaction, index, hasher) {
```

Comment:
```suggestion
    async signWithKey(address, transaction, index, hasher = TransactionHasher.getHashForSigning) {
```

Reply:
Since the `signWithKey` function has two optional parameters(index, hasher), I think it is clear to process in the code below rather than specifying the default parameter.

The`signWithKeys` function has a default parameter defined because only one parameter(hasher) is optional.